### DISCUSSION TITLE
OnBeat callback to enable creative coding? (Beat Tracking)

### DISCUSSION THREAD

**Discussion by tadev** [dlepaux/realtime-bpm-analyzer#76] [Q&A]
Hello, thank you for this library. I am currently evaluating different ways to animate elements based on live audio via microphone. Your project sparked my interest because it seems pretty accurate and fast.
Although I have not found a way to accurately use the detected BPM in js based animations, what I am trying to accomplish is to execute a function in the main animation loop every time the currently playing audio stream hits a beat.  Do you plan to support this or is there already a built in way?

**Comment by dlepaux** [comment]
Hi @tadev,

The library uses amplitude thresholding to compute the BPM. This means that basically we accumulate data and count peaks of bass frequencies to then determinate the BPM.

The usage you want to do is not compatible with the library because it is not designed to detect peaks in realtime.

I will study the topic and come back to you.
